### PR TITLE
Electron app

### LINKED ISSUE #1: [FEATURE] Features Requested by Jennie and Team 11
Per Jennie's email on 2024-03-02 and from our Team meeting on 2024-02-29:

```
I think you got this all from me, but just in case.

In / Out - what is the directionality of the counts - can an arrow or explantation be added to indicate?, and how does this work specifically with the vertical lines.

wheelchairs -> add to pedestrian count for now so it's not confusing

15 min clips - Rob mentioned (and I concur) that it would be good to have counts in 15 min clips as this is a standard increment used in counts

time conversion - will we need a way to compare the raw video time to the processed video time? in the event that the processing takes different amounts of time and buckets the data accordingly?

transparency - love the transparency idea for the paths

real time length of Allen clips - can you use the # of segments to estimate the real time length of the Allen clips that Rob and you took so that the counts are more meaningful

please save Allen St raw data - in case we need to count cars and trucks
```

In addition to this, Raif wants to save the x,y tracks into its own file and add an upload feature to upload a processed track file so that you don't have to go through and re-do a whole video if you close a session and want to do more analysis on it.

Written by Wendell

### LINKED ISSUE #2: [BUG] Web App Started From Command Line Does Not Process Correctly
Wendell says:

I don't remember exactly what the issue was, but something about the web app being autostarted did not properly process videos once they were uploaded (i.e. no terminal window was available to debug it either). It would work fine if the web app was fired from a terminal and the terminal window was left open for debugging purposes. @rolson24 need to investigate more

Video uploads successfully. However, upon drawing lines and clicking "Process Video", the loader would spin for a while and then immediately the progress bar would go to the end with "00:00:00" remaining on processing time left. And no count output would be produced.

### PR DESCRIPTION
**Pull request by rolson24**
This PR integrates the existing webapp into an electron wrapper so that it performs as a desktop app. Has been validated on Raif's Laptop to also work in Windows. (for install instructions, check the README)

### CODE REVIEW

File: BWCT_favicon.png
Comment (wuwendell) [inline BWCT_favicon.png:1]: I like this logo. great logo.

File: README.md
Comment (wuwendell) [inline README.md:1]: Sick `README.md` template

File: backend/BWCT_app.py
Comment (wuwendell) [inline backend/BWCT_app.py:1]: Insanely long file, we should try and split it up in the future, or at least define header comments for various sections of the code

TODO 24S

File: backend/templates/upload.html
Comment (wuwendell) [inline backend/templates/upload.html:1]: This file scares me.

File: backend/tracking/.DS_Store
Comment (wuwendell) [inline backend/tracking/.DS_Store:1]: nit: add `.DS_Store` to `.gitignore`.

File: backend/tracking/ConfTrack/ConfTrack
Comment (wuwendell) [inline backend/tracking/ConfTrack/ConfTrack:1]: Why the duplicate? Seems like it's just a python script without the `.py` file ending and you already have one in the same directory.

File: backend/tracking/ConfTrack/ConfTrack.py
Comment (wuwendell) [inline backend/tracking/ConfTrack/ConfTrack.py:1]: See comment above

File: backend/tracking/Impr_Assoc_Track/.DS_Store
Comment (wuwendell) [inline backend/tracking/Impr_Assoc_Track/.DS_Store:1]: Nit: remove `.DS_Store` from all git tracking and add to `.gitignore`

File: backend/tracking/LSTMTrack/LSTM_predictor.py
Comment (wuwendell) [inline backend/tracking/LSTMTrack/LSTM_predictor.py:1]: Nit: every file needs more header comments

File: index.html
Comment (wuwendell) [inline index.html:1]: This is the main html file for main page of electron app I assume?

File: backend/tracking/tests/ECSC-Walkway-4_track_output.txt
Comment (wuwendell) [inline backend/tracking/tests/ECSC-Walkway-4_track_output.txt:1]: Nit: feel like we should add test outputs to `.gitignore` too

File: backend/tracking/fast_reid/.github/FastReID-Logo.png
Comment (wuwendell) [inline backend/tracking/fast_reid/.github/FastReID-Logo.png:1]: Holy cow so many renamed files in this 60 commit PR from just renaming a directory... tis the pain of research code, or more like the pain of adding a complete other git repo into our own

**Review by wuwendell — APPROVED** [review]
Nice overall. Don't have to fix my PR comments right now if not desired, but let's def work on that over time. Lot of nits.

File: backend/BWCT_app.py
Comment (rolson24) [reply backend/BWCT_app.py:1]: Ya probably would be good to at least put the helper functions in their own file.

File: backend/templates/upload.html
Comment (rolson24) [reply backend/templates/upload.html:1]: yaa....

File: backend/tracking/fast_reid/.github/FastReID-Logo.png
Comment (rolson24) [reply backend/tracking/fast_reid/.github/FastReID-Logo.png:1]: Ya lol, I will try to come up with a better solution. I don't think they have packaged their repo though.

File: backend/tracking/ConfTrack/ConfTrack
Comment (rolson24) [reply backend/tracking/ConfTrack/ConfTrack:1]: Oh ya forgot to delete that

**Pull request merged**